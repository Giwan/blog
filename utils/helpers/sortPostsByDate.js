import { getDateNumber } from "../helpers";

/**
 * Sort posts by published date
 * @param {*} p1
 * @param {*} p2
 * @returns
 */
export const sortPostsByDate = (dateName) => (p1, p2) => {
    if (!(typeof dateName === "string" && dateName.length > 0)) {
        throw new Error("A valid dateName value is required");
    }

    const [date1, date2] = [p1, p2].map((date) => {
        const meta = date?.module?.meta;

        const dateString = meta[dateName];
        if (typeof dateString !== "string") {
            throw new Error(
                `Invalid ${dateName} date. Value of type string expected. Received: ${dateString}`
            );
        }
        return getDateNumber(dateString);
    });

    return date2 - date1;
};
