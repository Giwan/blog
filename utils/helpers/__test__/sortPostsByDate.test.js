import { sortPostsByDate } from "../sortPostsByDate";

describe("sort posts by date", () => {
    it("should sort the posts", () => {
        expect(() => sortPostsByDate()()).toThrowErrorMatchingSnapshot();
        const post1 = {
            module: {
                meta: {
                    published: "10-10-1980"
                }
            }
        }
        const post2 = {
            module: {
                meta: {
                    published: "12-10-1980"
                }
            }
        }
        expect(() => sortPostsByDate("createdDate")(post1, post2)).toThrowErrorMatchingSnapshot();
        expect(sortPostsByDate("published")(post1, post2)).toBe(2000000);
    });
});
