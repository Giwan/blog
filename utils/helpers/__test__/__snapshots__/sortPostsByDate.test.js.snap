// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sort posts by date should sort the posts 1`] = `"A valid dateName value is required"`;

exports[`sort posts by date should sort the posts 2`] = `"Invalid createdDate date. Value of type string expected. Received: undefined"`;
